drivers/tsl4531x/tsl4531x_saul.c

```diff
@@ -27,7 +27,7 @@
 static int _read(const void *dev, phydat_t *data)
 {
     memset(data, 0, sizeof(phydat_t));
-    data->val[0] = tsl4531x_simple_read((const tsl4531x_t *)dev);
+    data->val[0] = tsl4531x_simple_read(*(tsl4531x_t **)dev);
```

Comment:
Replace by `(tsl4531x_t **)` by `(tsl4531x_t * const*)`, otherwise you are casting to an incompatible pointer type (you can check it by compiling with -Wcast-qual).
The end result is the same, of course.

Reply:
will do, thanks